mlflow/openai/utils.py

```diff
@@ -92,10 +111,81 @@ def request(*args, **kwargs):
         if url.endswith("/chat/completions"):
             messages = json.loads(kwargs.get("data")).get("messages")
             return _mock_chat_completion_response(content=json.dumps(messages))
+        elif url.endswith("/completions"):
+            prompt = json.loads(kwargs.get("data")).get("prompt")
+            return _mock_completion_response(content=json.dumps(prompt))
         elif url.endswith("/embeddings"):
             inp = json.loads(kwargs.get("data")).get("input")
             return _mock_embeddings_response(len(inp) if isinstance(inp, list) else 1)
         else:
             return original(*args, **kwargs)
 
     return _mock_request(new=request)
+
+
+def _get_task_model_params(model: Dict[str, Any], params: Dict[str, Any] = None):
+    model_dict = model.copy()
+    task = model_dict.pop("task", None)
+    if params:
+        if any(params.keys() - model_dict.keys()):
+            raise mlflow.MlflowException.invalid_parameter_value(
+                f"Providing any of {list(model_dict.keys())} as parameters is not allowed.",
+            )
+
+    return task, model_dict, params
+
+
+def _validate_task_model_params(model: Dict[str, Any], schema: ParamSchema = None):
+    if schema:
+        if any(param.name in model.keys() for param in schema.params):
+            raise mlflow.MlflowException.invalid_parameter_value(
+                f"Providing any of {','.join(model.keys())} as parameters is not allowed.",
+            )
+
+
+class _OAITokenHolder:
+    def __init__(self, api_type):
+        import openai
+
+        self.api_type = api_type
+        self.api_token = None
+
+        if self.api_type in ("azure_ad", "azuread"):
+            try:
+                from azure.identity import DefaultAzureCredential
+            except ImportError:
+                raise mlflow.MlflowException(
+                    "Using API type ``azure_ad`` or ``azuread`` requires the package"
+                    " ``azure-identity`` to be installed."
+                )
+            self.credential = DefaultAzureCredential()
+        else:
+            self.openai_key_available = openai.api_key or "OPENAI_API_KEY" in os.environ
```

Comment:
We should avoid conditionally defining ana attribute. That can lead to an error.